src/eliminate_contiguous.cpp

```diff
@@ -70,53 +70,79 @@ static bool try_compute_shape(instruction_ref ins,
     return try_compute_shape(ins, inputs, mods);
 }
 
-void eliminate_contiguous::apply(module& m) const
+template <class F>
+static void remove_contiguous(const std::string& op_name, module& m, F f)
 {
-    std::vector<instruction_ref> const_instruction;
+    auto last = std::prev(m.end());
+    std::vector<instruction_ref> const_instructions;
 
     for(auto ins : iterator_for(m))
     {
         // return instruction should have inputs with standard shape
         if(ins->name() == "@return")
             continue;
 
+        if(ins != last and ins->outputs().empty())
```

Comment:
Shouldn't this be eliminated by deadcode elimination ?

Reply:
It will be, but this will skip code that will be eliminated by dead code elimination.